<a href="https://colab.research.google.com/github/princessivy/Analyzing_CSGO_SkinPrices/blob/main/Scraping.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Imports & Installs

In [ ]:
!pip install bs4 --quiet

In [1]:
import pandas as pd
import requests
import json
from bs4 import BeautifulSoup

# Nutzung BeautifulSoup

In [3]:
r = requests.get("https://www.immowelt.de/liste/wuerzburg/wohnungen/mieten?d=true&sd=DESC&sf=RELEVANCE&sp=1")

In [4]:
# check, ob alles gut gegangen ist (200 = OK)
r.status_code

200

In [5]:
# check Typ
r.headers['content-type']

'text/html'

In [6]:
 # check Encoding
 r.encoding

'ISO-8859-1'

In [7]:
r.encoding = 'utf-8'

In [8]:
# print Seite als HTML
html = r.text
html

'<!doctype html>\n<html lang="de">\n  <head>\n    <meta data-react-helmet="true" name="google" content="notranslate"/><meta data-react-helmet="true" charset="utf-8"/><meta data-react-helmet="true" name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no, user-scalable=no"/><meta data-react-helmet="true" name="description" content="Wohnung mieten in Würzburg. Wohnimmobilien zur Miete findest du bei Immowelt ⇒ Jetzt dank großer Auswahl Wunschimmobilie finden!"/><meta data-react-helmet="true" name="robots" content="noindex"/><meta data-react-helmet="true" property="og:title" content="Wohnung mieten in Würzburg"/><meta data-react-helmet="true" property="og:description" content="Wohnung mieten in Würzburg. Wohnimmobilien zur Miete findest du bei Immowelt ⇒ Jetzt dank großer Auswahl Wunschimmobilie finden!"/><meta data-react-helmet="true" property="og:image" content="//media-static.immowelt.org/app_themes/mid_0_rwd/image/logo/og-img_iwde.png"/><meta data-react-helmet="t

In [9]:
print(html)

<!doctype html>
<html lang="de">
  <head>
    <meta data-react-helmet="true" name="google" content="notranslate"/><meta data-react-helmet="true" charset="utf-8"/><meta data-react-helmet="true" name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no, user-scalable=no"/><meta data-react-helmet="true" name="description" content="Wohnung mieten in Würzburg. Wohnimmobilien zur Miete findest du bei Immowelt ⇒ Jetzt dank großer Auswahl Wunschimmobilie finden!"/><meta data-react-helmet="true" name="robots" content="noindex"/><meta data-react-helmet="true" property="og:title" content="Wohnung mieten in Würzburg"/><meta data-react-helmet="true" property="og:description" content="Wohnung mieten in Würzburg. Wohnimmobilien zur Miete findest du bei Immowelt ⇒ Jetzt dank großer Auswahl Wunschimmobilie finden!"/><meta data-react-helmet="true" property="og:image" content="//media-static.immowelt.org/app_themes/mid_0_rwd/image/logo/og-img_iwde.png"/><meta data-react-helmet="true"

In [10]:
soup = BeautifulSoup(html,'html.parser')

In [11]:
soup.title

<title data-react-helmet="true">Wohnung mieten in Würzburg</title>

In [12]:
# gibt nur den ersten zurück
soup.title.string

'Wohnung mieten in Würzburg'

In [13]:
soup.title.string.parent

<title data-react-helmet="true">Wohnung mieten in Würzburg</title>

In [14]:
h2 = []
for tag in soup.findAll('h2'):
  h2.append(tag.string)
  #print(tag.string)

In [15]:
h2

['3-Zimmer-Wohnung  90 m² Wfl. im 2. Stock - Exklusiv',
 '3-Zi- Penthaus-Wohnung Holzbau, Würzburg Hubland',
 'Sonnige 3 Zi-Whg. mit Blick über Versbach',
 'Schicke 3-Zimmer Wohnung mit Glaserker und Festungsblick',
 '2 Zimmer Neubauwohnung, 1. OG, barrierefrei mit Aufzug',
 'Eine Wohnung nach Ihrem Geschmack! Stillvolles Wohnen mit Garten',
 'Renovierte 2,5 ZW mit 2 Balkonen, Aufzug und TG-Stellplatz.',
 'Weitblick 3,5 Zimmer helles und ruhiges wohnen mit EBK, Balkon, Tageslichtbad mit Wanne, Garten u.v.m.',
 'Wohnen im beliebten Frauenland, nähe Wittelsbacher Platz.',
 'Schöne, möblierte 2,5-ZW, ca. 79 qm, mit Balkon in Würzburg / Nähe Unikliniken',
 'Ihr neues Zuhause - 2,5 Zimmer im 2. Obergeschoss mit Ankleide, Balkon und Keller',
 "Luxuriöse 3-Zi.-Wohnung Nähe Residenz (Platz'scher Garten)",
 '+++ Zauberhafte komplett neu sanierte 2-ZW mit Balkon in Unterdürrbach  +++',
 'Attraktive 2 Zimmer City-Wohnung',
 'Charmantes 1 Zimmer Apartment im Frauenland - beste Lage - mit großem Ba

In [ ]:
'''for tag in soup.findAll('div', {'class':"KeyFacts-efbce"}):
  print(tag)'''

'for tag in soup.findAll(\'div\', {\'class\':"KeyFacts-efbce"}):\n  print(tag)'

In [16]:
price = []
for tag in soup.findAll('div', {'data-test':"price"}):
  price.append(tag.string)
  #print(tag.string)

In [17]:
qm = []
for tag in soup.findAll('div', {'data-test':"area"}):
  qm.append(tag.string)
  #print(tag.string)

In [18]:
zimmer = []
for tag in soup.findAll('div', {'data-test':"rooms"}):
  zimmer.append(tag.string)
  #print(tag.string)

In [ ]:
'''for tag in soup.findAll('div', {'class':"IconFact-e8a23"}):
  print(tag)'''

'for tag in soup.findAll(\'div\', {\'class\':"IconFact-e8a23"}):\n  print(tag)'

In [19]:
adresse = []
for tag in soup.findAll('div', {'class':"IconFact-e8a23"}):
  if tag.i.text == "location":
    #print(tag.span.text)
    adresse.append(tag.span.text)

In [20]:
besonderes = []
for tag in soup.findAll('div', {'class':"IconFact-e8a23"}):
  if tag.i.text == "check":
    #print(tag.span.text)
    besonderes.append(tag.span.text)

In [21]:
df = pd.DataFrame({'Text': h2, 'Adresse':adresse, 'Preis':price, 'Quadratmeter':qm, 'Anzahl Zimmer':zimmer, 'Besonderes':besonderes})

In [22]:
df

Text  \
0   3-Zimmer-Wohnung  90 m² Wfl. im 2. Stock - Exk...   
1    3-Zi- Penthaus-Wohnung Holzbau, Würzburg Hubland   
2           Sonnige 3 Zi-Whg. mit Blick über Versbach   
3   Schicke 3-Zimmer Wohnung mit Glaserker und Fes...   
4   2 Zimmer Neubauwohnung, 1. OG, barrierefrei mi...   
5   Eine Wohnung nach Ihrem Geschmack! Stillvolles...   
6   Renovierte 2,5 ZW mit 2 Balkonen, Aufzug und T...   
7   Weitblick 3,5 Zimmer helles und ruhiges wohnen...   
8   Wohnen im beliebten Frauenland, nähe Wittelsba...   
9   Schöne, möblierte 2,5-ZW, ca. 79 qm, mit Balko...   
10  Ihr neues Zuhause - 2,5 Zimmer im 2. Obergesch...   
11  Luxuriöse 3-Zi.-Wohnung Nähe Residenz (Platz's...   
12  +++ Zauberhafte komplett neu sanierte 2-ZW mit...   
13                   Attraktive 2 Zimmer City-Wohnung   
14  Charmantes 1 Zimmer Apartment im Frauenland - ...   
15            +++  Bereits V E R M I E T E T !!!  +++   
16  3-Zimmer-Wohnung in traumhafter Lage mit Balko...   
17  **großzügig, helle und hochwertig ruhig gelege...   
18  Hochwertig möblierte, kernsanierte 2-Zimmerwoh...   
19  90004      2,5 Zimmerwohnung in Würzburg-Lengfeld   

                                             Adresse        Preis  \
0    Dr  -Georg-Fuchs-Str. 12, Würzburg (Frauenland)      1.225 €   
1   Norbert-Glanzberg-Str. 14, Würzburg (Frauenland)      1.628 €   
2                                Würzburg (Versbach)        670 €   
3                Gartenstraße 9, Würzburg (Sanderau)        990 €   
4      Winterhäuser Str. 24, Würzburg (Heidingsfeld)        666 €   
5   Andreas Grieser Str. 32, Würzburg (Heidingsfeld)      1.550 €   
6            Pilziggrundstr. 56, Würzburg (Lengfeld)        830 €   
7                            Würzburg (Steinbachtal)      1.130 €   
8                              Würzburg (Frauenland)        600 €   
9                                Würzburg (Versbach)        890 €   
10   Jakob-Riedinger-Straße 4, Würzburg (Frauenland)        850 €   
11      Rottendorfer Straße 2 c, Würzburg (Altstadt)      1.190 €   
12                          Würzburg (Unterdürrbach)        550 €   
13                               Würzburg (Altstadt)        720 €   
14                             Würzburg (Frauenland)        495 €   
15                           Würzburg (Oberdürrbach)  auf Anfrage   
16                 Hübergasse 2, Würzburg (Altstadt)      1.100 €   
17                               Würzburg (Lengfeld)        820 €   
18                               Würzburg (Grombühl)        990 €   
19                               Würzburg (Lengfeld)        720 €   

   Quadratmeter Anzahl Zimmer                                      Besonderes  
0         90 m²         3 Zi.  Personenaufzug, Einbauküche, barrierefrei, ...  
1     121.51 m²         3 Zi.         Personenaufzug, Tiefgarage, Balkon, ...  
2         82 m²         3 Zi.    Einbauküche, Balkon, Gartenmitbenutzung, ...  
3      79.76 m²         3 Zi.      Personenaufzug, Balkon, Bad mit Wanne, ...  
4       55.5 m²         2 Zi.  Personenaufzug, Einbauküche, barrierefrei, ...  
5     124.49 m²         3 Zi.  Personenaufzug, Einbauküche, barrierefrei, ...  
6         76 m²       2.5 Zi.             Personenaufzug, Balkon, Garten, ...  
7         95 m²       3.5 Zi.             vermietet, Einbauküche, Balkon, ...  
8         50 m²         2 Zi.         barrierefrei, saniert, Erdgeschoss, ...  
9         79 m²       2.5 Zi.             renoviert, Einbauküche, Balkon, ...  
10        68 m²       2.5 Zi.               Einbauküche, Balkon, saniert, ...  
11        92 m²         3 Zi.    Personenaufzug, Einbauküche, Tiefgarage, ...  
12        48 m²         2 Zi.    Balkon, saniert, Frei / nicht vermietet, ...  
13        60 m²         2 Zi.         Einbauküche, Balkon, Bad mit Wanne, ...  
14        33 m²         1 Zi.               Einbauküche, Balkon, Kelleranteil  
15        21 m²         1 Zi.        Einbauküche, Erdgeschoss, renoviert, ...  
16    107.02 m²         3 Zi.            

# Nutzung Requests (API only)

In [23]:
r = requests.get("https://api.immowelt.com/residentialsearch/v1/searches")

In [24]:
r

<Response [403]>

Response ist 403, heißt, dass der Zugriff auf die API nicht so einfach geht, wie der Zugriff auf das HTML

In [29]:
url = "https://api.immowelt.com/residentialsearch/v1/searches"

payload = json.dumps({
  "estateType": "APARTMENT",
  "distributionTypes": [
    "RENT",
    "LEASE"
  ],
  "estateSubtypes": [],
  "locationIds": [
    4945
  ],
  "featureFilters": [],
  "excludedFeatureFilters": [],
  "primaryPrice": {
    "min": None,
    "max": None
  },
  "primaryArea": {
    "min": None,
    "max": None
  },
  "areas": [
    {
      "areaType": "PLOT_AREA",
      "min": None,
      "max": None
    }
  ],
  "rooms": {
    "min": None,
    "max": None
  },
  "constructionYear": {
    "min": None,
    "max": None
  },
  "geoRadius": {
    "radius": None,
    "point": {
      "lat": 49.78471551452244,
      "lon": 9.940873229009464
    }
  },
  "zipCode": None,
  "sort": {
    "direction": "DESC",
    "field": "RELEVANCE"
  },
  "immoItemTypes": [
    "ESTATE",
    "PROJECT"
  ],
  "paging": {
    "size": 20,
    "page": 0
  }
})
headers = {
  'Accept': 'application/json, text/plain, */*',
  'Origin': 'https://www.immowelt.de',
  'Content-Type': 'application/json',
  'Authorization': 'Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IjAwMzIwMjAwNjExIn0.eyJzY29wZSI6WyJzZXJ2aWNlcyJdLCJpc3MiOiJJV1QiLCJleHAiOjE2NTI3MDY1MDUsImNyZWF0aW9uX29mZnNldF9kYXRlX3RpbWUiOiIyMDIyLTA1LTE2VDE0OjA4OjI1Ljc5MDcwNSswMjowMCIsImF1dGhvcml0aWVzIjpbInB1YmxpYyIsInJlY29tbWVuZGVyIiwiY2xpY2tzdHJlYW1lciJdLCJqdGkiOiJ0SmVuUEctWXRVNTA1Y1hmSGxrUGN5dmNaLWMiLCJ0ZW5hbnQiOiJpbW1vd2VsdCIsImNsaWVudF9pZCI6InJlc2lkZW50aWFsLXNlYXJjaC11aSJ9.MjQoiOMnATZafevFO1ig7m9IoemEbZVi8iqUoShjOH6fTTRe3y5jD1zyFskTNPjfXNNrprDFm7fv7SOOMP5R16VxiAWa8OmB7DBz_Jdz4SG_bjH5C1uqX_ru2bxVMyIbjPAtwROXb7mDEnMIy4rR1eXUvDXXS18W7UCE5CN1S6dsCWDY1WVAJYjsnlTZnnds8AQtrD12eabCMjaU2ujqP10AXv9S4KIVxo5rouTTpiinvR-3D4ssqkFw__NDYUKdjDH6zc1pUsMl5frUifhIgTS_RG0F9wXYLqvtAh_agqkGz52DfFqMTOj1enR-vJbjHBYVUSW3U4jNEBo-aKzqdw',
  'Referer': 'https://www.immowelt.de/',
  'Content-Length': '577',
  'Host': 'api.immowelt.com',
  'Accept-Language': 'de-DE,de;q=0.9',
  'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15',
  'Accept-Encoding': 'gzip, deflate, br',
  'Connection': 'keep-alive'
}

response0 = requests.request("POST", url, headers=headers, data=payload)

print(response0.text)


{"sort":{"direction":"DESC","field":"RELEVANCE"},"pagesCount":7,"data":[{"id":"estate_285b4fc4-d626-4ebb-a71e-3f81b3a52cc1","itemType":"ESTATE","gok":"285b4fc4-d626-4ebb-a71e-3f81b3a52cc1","locationIds":[525728,526018,528418,496008,4945,113848,122156,121726],"lowestLevelLocationId":496008,"place":{"point":{"lat":49.789626,"lon":9.962649,"geohash":"u0z54xytddcx","fragment":true},"postcode":"97074","city":"Würzburg","district":"Frauenland","street":"Dr  -Georg-Fuchs-Str.","houseNumber":"12","geoPointIsAnonymized":false},"estateTypes":["APARTMENT"],"estateSubtypes":[],"distributionType":"RENT","title":"3-Zimmer-Wohnung  90 m² Wfl. im 2. Stock - Exklusiv","prices":[{"amountMin":1225.0,"amountMax":1225.0,"type":"COLD_RENT","currency":"EUR","attributes":["FIXED_PRICE"]},{"amountMin":1525.0,"amountMax":1525.0,"type":"RENT_INCLUDING_HEATING","currency":"EUR","attributes":["INCLUSIVE_ADDITIONAL_COSTS","FIXED_PRICE"]},{"amountMin":90.0,"amountMax":90.0,"currency":"EUR","attributes":[]}],"areas":

In [30]:
page0_wuerzburg = response0.text

In [31]:
page0_wuerzburg

'{"sort":{"direction":"DESC","field":"RELEVANCE"},"pagesCount":7,"data":[{"id":"estate_285b4fc4-d626-4ebb-a71e-3f81b3a52cc1","itemType":"ESTATE","gok":"285b4fc4-d626-4ebb-a71e-3f81b3a52cc1","locationIds":[525728,526018,528418,496008,4945,113848,122156,121726],"lowestLevelLocationId":496008,"place":{"point":{"lat":49.789626,"lon":9.962649,"geohash":"u0z54xytddcx","fragment":true},"postcode":"97074","city":"Würzburg","district":"Frauenland","street":"Dr  -Georg-Fuchs-Str.","houseNumber":"12","geoPointIsAnonymized":false},"estateTypes":["APARTMENT"],"estateSubtypes":[],"distributionType":"RENT","title":"3-Zimmer-Wohnung  90 m² Wfl. im 2. Stock - Exklusiv","prices":[{"amountMin":1225.0,"amountMax":1225.0,"type":"COLD_RENT","currency":"EUR","attributes":["FIXED_PRICE"]},{"amountMin":1525.0,"amountMax":1525.0,"type":"RENT_INCLUDING_HEATING","currency":"EUR","attributes":["INCLUSIVE_ADDITIONAL_COSTS","FIXED_PRICE"]},{"amountMin":90.0,"amountMax":90.0,"currency":"EUR","attributes":[]}],"areas"

In [35]:
url = "https://api.immowelt.com/residentialsearch/v1/searches"

payload = json.dumps({
  "estateType": "APARTMENT",
  "distributionTypes": [
    "RENT",
    "LEASE"
  ],
  "estateSubtypes": [],
  "locationIds": [
    4945
  ],
  "featureFilters": [],
  "excludedFeatureFilters": [],
  "primaryPrice": {
    "min": None,
    "max": None
  },
  "primaryArea": {
    "min": None,
    "max": None
  },
  "areas": [
    {
      "areaType": "PLOT_AREA",
      "min": None,
      "max": None
    }
  ],
  "rooms": {
    "min": None,
    "max": None
  },
  "constructionYear": {
    "min": None,
    "max": None
  },
  "geoRadius": {
    "radius": None,
    "point": {
      "lat": 49.78471551452244,
      "lon": 9.940873229009464
    }
  },
  "zipCode": None,
  "sort": {
    "direction": "DESC",
    "field": "RELEVANCE"
  },
  "immoItemTypes": [
    "ESTATE",
    "PROJECT"
  ],
  "paging": {
    "size": 20,
    "page": 1
  }
})
headers = {
  'Accept': 'application/json, text/plain, */*',
  'Origin': 'https://www.immowelt.de',
  'Content-Type': 'application/json',
  'Authorization': 'Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IjAwMzIwMjAwNjExIn0.eyJzY29wZSI6WyJzZXJ2aWNlcyJdLCJpc3MiOiJJV1QiLCJleHAiOjE2NTI3MDY1MDUsImNyZWF0aW9uX29mZnNldF9kYXRlX3RpbWUiOiIyMDIyLTA1LTE2VDE0OjA4OjI1Ljc5MDcwNSswMjowMCIsImF1dGhvcml0aWVzIjpbInB1YmxpYyIsInJlY29tbWVuZGVyIiwiY2xpY2tzdHJlYW1lciJdLCJqdGkiOiJ0SmVuUEctWXRVNTA1Y1hmSGxrUGN5dmNaLWMiLCJ0ZW5hbnQiOiJpbW1vd2VsdCIsImNsaWVudF9pZCI6InJlc2lkZW50aWFsLXNlYXJjaC11aSJ9.MjQoiOMnATZafevFO1ig7m9IoemEbZVi8iqUoShjOH6fTTRe3y5jD1zyFskTNPjfXNNrprDFm7fv7SOOMP5R16VxiAWa8OmB7DBz_Jdz4SG_bjH5C1uqX_ru2bxVMyIbjPAtwROXb7mDEnMIy4rR1eXUvDXXS18W7UCE5CN1S6dsCWDY1WVAJYjsnlTZnnds8AQtrD12eabCMjaU2ujqP10AXv9S4KIVxo5rouTTpiinvR-3D4ssqkFw__NDYUKdjDH6zc1pUsMl5frUifhIgTS_RG0F9wXYLqvtAh_agqkGz52DfFqMTOj1enR-vJbjHBYVUSW3U4jNEBo-aKzqdw',
  'Referer': 'https://www.immowelt.de/',
  'Content-Length': '577',
  'Host': 'api.immowelt.com',
  'Accept-Language': 'de-DE,de;q=0.9',
  'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15',
  'Accept-Encoding': 'gzip, deflate, br',
  'Connection': 'keep-alive'
}

response1 = requests.request("POST", url, headers=headers, data=payload)

print(response1.text)


{"sort":{"direction":"DESC","field":"RELEVANCE"},"data":[{"id":"estate_98c7cf11-ba21-4956-99b1-603fd92bcc08","itemType":"ESTATE","gok":"98c7cf11-ba21-4956-99b1-603fd92bcc08","locationIds":[530510,526018,528418,496026,4945,113848,122156,121726],"lowestLevelLocationId":496026,"place":{"point":{"lat":49.791238,"lon":9.905827,"geohash":"u0z532jnyrrz","fragment":true},"postcode":"97082","city":"Würzburg","district":"Zellerau","street":"Spessartstraße","geoPointIsAnonymized":false},"estateTypes":["APARTMENT"],"estateSubtypes":[],"distributionType":"RENT","title":"Exklusives Angebot: Modern sanierte 2,5-Zimmer-Wohnung mit Balkon","prices":[{"amountMin":960.0,"amountMax":960.0,"type":"RENT_INCLUDING_HEATING","currency":"EUR","attributes":["INCLUSIVE_ADDITIONAL_COSTS","FIXED_PRICE"]},{"amountMin":720.0,"amountMax":720.0,"type":"COLD_RENT","currency":"EUR","attributes":["FIXED_PRICE"]},{"amountMin":65.0,"amountMax":65.0,"currency":"EUR","attributes":[]}],"areas":[{"sizeMin":66.0,"sizeMax":66.0,"

In [37]:
page1_wuerzburg = response1.text

In [ ]:
# bayern

In [40]:
json_wue = json.loads(page0_wuerzburg)

In [43]:
json_wue['data'][0]

{'areas': [{'sizeMax': 90.0, 'sizeMin': 90.0, 'type': 'LIVING_AREA'}],
 'broker': {'companyName': None,
  'guid': 3472276,
  'immoweltExclusive': False,
  'logoUri': None,
  'partnerAward': None,
  'sellerType': 'EAZ_PRIVATE'},
 'constructionYear': 2019,
 'distributionType': 'RENT',
 'estateSubtypes': [],
 'estateTypes': ['APARTMENT'],
 'features': ['PASSENGER_LIFT',
  'FITTED_KITCHEN',
  'BARRIER_FREE',
  'BALCONY',
  'NEW_BUILDING',
  'BATH_WITH_TUB',
  'CELLAR_SHARE',
  'WHEELCHAIR_ACCESSIBLE',
  'PARKING_AREA',
  'BATH_WITH_WINDOW'],
 'gok': '285b4fc4-d626-4ebb-a71e-3f81b3a52cc1',
 'id': 'estate_285b4fc4-d626-4ebb-a71e-3f81b3a52cc1',
 'isNew': True,
 'itemType': 'ESTATE',
 'locationIds': [525728, 526018, 528418, 496008, 4945, 113848, 122156, 121726],
 'lowestLevelLocationId': 496008,
 'onlineId': '2pb634z',
 'pictures': [{'description': '600_B91C3DA8005A4778871D39B72FDE5397',
   'imageUri': 'https://ms.immowelt.org/c1a2d246-4be5-4180-aa42-06ea74b2afaa/9f3dd3aa-4100-4ea5-bfd9-3311a2

In [47]:
dataframe = pd.DataFrame()

dataframe = dataframe.append(json_wue['data'][0], ignore_index=True)

In [49]:
dataframe.columns

Index(['id', 'itemType', 'gok', 'locationIds', 'lowestLevelLocationId',
       'place', 'estateTypes', 'estateSubtypes', 'distributionType', 'title',
       'prices', 'areas', 'primaryArea', 'primaryPrice', 'roomsMin',
       'roomsMax', 'constructionYear', 'timestamp', 'isNew', 'features',
       'pictures', 'presentationProducts', 'broker', 'onlineId',
       'relatedItems'],
      dtype='object')

In [52]:
#titel aka. text
json_wue['data'][0]['title']

'3-Zimmer-Wohnung  90 m² Wfl. im 2. Stock - Exklusiv'

In [57]:
# Adresse
json_wue['data'][0]['place']

{'city': 'Würzburg',
 'district': 'Frauenland',
 'geoPointIsAnonymized': False,
 'houseNumber': '12',
 'point': {'fragment': True,
  'geohash': 'u0z54xytddcx',
  'lat': 49.789626,
  'lon': 9.962649},
 'postcode': '97074',
 'street': 'Dr  -Georg-Fuchs-Str.'}

In [58]:
json_wue['data'][0]['place']['city']

'Würzburg'

In [59]:
json_wue['data'][0]['place']['district']

'Frauenland'

In [60]:
json_wue['data'][0]['place']['houseNumber']

'12'

In [62]:
json_wue['data'][0]['place']['point']['lat']

49.789626

In [63]:
json_wue['data'][0]['place']['point']['lon']

9.962649

In [64]:
json_wue['data'][0]['place']['postcode']

'97074'

In [65]:
json_wue['data'][0]['place']['street']

'Dr  -Georg-Fuchs-Str.'

In [66]:
# Preis
json_wue['data'][0]['prices']

[{'amountMax': 1225.0,
  'amountMin': 1225.0,
  'attributes': ['FIXED_PRICE'],
  'currency': 'EUR',
  'type': 'COLD_RENT'},
 {'amountMax': 1525.0,
  'amountMin': 1525.0,
  'attributes': ['INCLUSIVE_ADDITIONAL_COSTS', 'FIXED_PRICE'],
  'currency': 'EUR',
  'type': 'RENT_INCLUDING_HEATING'},
 {'amountMax': 90.0, 'amountMin': 90.0, 'attributes': [], 'currency': 'EUR'}]

In [70]:
# Kaltmiete
json_wue['data'][0]['prices'][0]['amountMax']

1225.0

In [71]:
# Warmmiete
json_wue['data'][0]['prices'][1]['amountMax']

1525.0

In [73]:
# QM
json_wue['data'][0]['primaryArea']['sizeMax']

90.0

In [74]:
# Anzahl Zimmer
json_wue['data'][0]['roomsMax']

3.0

In [75]:
# Besonderes
json_wue['data'][0]['features']


['PASSENGER_LIFT',
 'FITTED_KITCHEN',
 'BARRIER_FREE',
 'BALCONY',
 'NEW_BUILDING',
 'BATH_WITH_TUB',
 'CELLAR_SHARE',
 'WHEELCHAIR_ACCESSIBLE',
 'PARKING_AREA',
 'BATH_WITH_WINDOW']